### PR TITLE
Support new database schema

### PR DESCRIPTION
**Pull request by om-sharma**
This PR migrates the data from `entries` table to `entry` and `item` table.

A scheduler thread reads the entries table and copies data into entry and item table. while migration is in progress the minting the data happens only in entries table. After the migration is complete the next app restart will start the scheduled thread for migration and new entries will be minted in both old and new tables.

### CODE REVIEW

File: src/main/java/uk/gov/mint/DataReplicationTask.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private EntryStore entryStore;
+
+    public DataReplicationTask(EntriesUpdateDAO entriesUpdateDAO, EntryStore entryStore) {
+        this.entriesUpdateDAO = entriesUpdateDAO;
+        this.entryStore = entryStore;
+    }
+
+    @Override
+    public void run() {
+        for (List<OldSchemaEntry> entries; !(entries = entriesUpdateDAO.read(lastReadID())).isEmpty(); ) {
```
Comment (philandstuff) [inline src/main/java/uk/gov/mint/DataReplicationTask.java:21]: does this need to be a `List`? Would it be better as a lazy `Iterable`?

File: src/main/java/uk/gov/mint/Item.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Item item = (Item) o;
+
+        return !(sha256hex != null ? !sha256hex.equals(item.sha256hex) : item.sha256hex != null);
```
Comment (philandstuff) [inline src/main/java/uk/gov/mint/Item.java:41]: :+1: love this equals() function.

If I were picky, I'd say we don't need the null checks as `sha256hex` can't be null. but I don't think it matters too much.

**Comment by om-sharma** [comment]
@philandstuff: If like, you can merge this PR and I will create a new PR for the suggestions here. 
I tried to use Iterable but string template causing `ClassCastException` because it tries to cast subclass of Iterable in Collection.

**Comment by om-sharma** [comment]
Raise new PR https://github.com/openregister/mint/pull/71 for suggested changes.

**Pull request merged**